src/main/java/commands/AddSalesCommand.java

```diff
@@ -41,20 +45,37 @@ public AddSalesCommand(String[] argsAuthorMangaQtyPrice) {
     public void execute(Ui ui, AuthorList authorList) throws TantouException {
         String authorName = argsAuthorMangaQtyPrice[AUTHOR_INDEX];
         String mangaName = argsAuthorMangaQtyPrice[MANGA_INDEX];
-        int quantitySold = Integer.parseInt(argsAuthorMangaQtyPrice[QUANTITY_INDEX]);
-        double unitPrice = Double.parseDouble(argsAuthorMangaQtyPrice[PRICE_INDEX]);
+        int quantitySold = 0;
+        double unitPrice = Double.parseDouble(argsAuthorMangaQtyPrice[PRICE_INDEX]);;
+        try {
+            quantitySold = Integer.parseInt(argsAuthorMangaQtyPrice[QUANTITY_INDEX]);
+            if (quantitySold >= 1000000000) {
+                throw new QuantityTooLargeException();
+            }
```

Comment:
Appreciate the catch if the user tries to enter an unrealistic input, maybe we can put the double parse inside the try block as well?